src/smt/smt_engine.cpp

```diff
@@ -1130,6 +1118,27 @@ void SmtEngine::setLogic(const std::string& s)
   SmtScope smts(this);
   try {
     setLogic(LogicInfo(s));
+
+    // dump out a set-logic command
+    if (Dump.isOn("benchmark"))
```

Comment:
Can you comment on the change(s) in behavior of moving this code?

Reply:
I had a problem where declare-var command sometimes gets printed before the set-logic command (which is printed in SmtEngine::finishInit function). Because of that, CVC4, printed warnings about not setting the logic and some tests failed. I moved the code from finishInit to this function to ensure that set-logic command is printed as soon as it's encountered.

Reply:
Right, (I think) I agree with this change however it does effect the default behavior for this and other use cases.

For example, another consequence is that we don't print any `set-logic` when dumping `raw-benchmark` for benchmarks without a set-logic command, whereas previously the code always force-printed a `(set-logic ALL)`.

Reply:
After discussion on devs, this block of code should be broken into two: 
- The case where `raw-benchmark` is true, should be moved here, as you did,
- The case where `raw-benchmark` is false should remain where it was (move it back to its original place).